### DISCUSSION TITLE
UBL and Youtube

### DISCUSSION THREAD

**Discussion by Pierre7855** [uBlockOrigin/uBOL-home#191] [Q&A]
Hello,

For Youtube I set Ublock origin Lite with all filters checked ON, and the main filter on "Complete".
Result  : it is difficult to see videos on Youtube (it takes a long time to load and it does not go directly to the main video.

I have two questions : 
1) in UBL, what filter do you advise me to choose on Youtube ?
2) I have noticed that UBO is working very well. Why this difference on Youtube between UBL and UBO ?

**Comment by stephenhawk8054** [comment]
> all filters checked ON

Don't do that. Reset uBOL to default settings + "Complete" mode (might need to uninstall and reinstall it), turn off all other extensions (except uBOL), restart the browser and test again.

**Reply by garry-ut99** [reply]
Depends on what exactly OP meant by "all filters checked ON": whether really all: including all regional filter lists, or only main lists. But if the former, then yes, he shouldn't enable all regional lists.

**Reply by Yuki2718** [reply]
"Optimal" mode is fine too for YT.

**Comment by Pierre7855** [comment]
Hello,
Here are my filters on UBL (when I talk about "all filters") : 

![screenshot-10-55-26](https://github.com/user-attachments/assets/d7187535-0d09-4451-8c4d-40c36fcb6622)

Now, I have just uninstalled and re-install UBL and I have selected only UBL "default settings" + "Complete mode" : 

![screenshot-11-06-10](https://github.com/user-attachments/assets/ed1c7f17-5b9f-43bf-838c-2695ebbc5066)

But the problem is always here  on Youtube  : 
1) when I click on a Youtube video on Youtube site, it takes a long time to load.
2) I do not access directly to the main video (it indicates first the time of each ad video before the main video while loading all the time).

**Reply by stephenhawk8054** [reply]
Can you go to `chrome://extensions` page (type in URL bar) and screenshot it here?

**Comment by Pierre7855** [comment]
Here are all my Chrome extensions : 

![screenshot-11-21-48](https://github.com/user-attachments/assets/ed2cf61a-11b7-43e0-8036-b40718fbe845)

**Reply by stephenhawk8054** [reply]
Can you turn off IDM, restart the browser and test again?

**Reply by garry-ut99** [reply]
Try again with all extensions disabled except uBOL.

Besides, this is Discussions, not Issues, usually you should reply in "Write a reply" field, and not in "Add your answer here" field, otherwise it will be hard to read the whole conversation.

<hr>

By the way: some weird github (or not github) lag, Stephen's comment didn't show on my screen until I pressed "Reply" , then the both comments have appeared at the same time on my screen, despite Stephen's comment was created 4 minutes earlier and was not visible to me....

**Reply by Pierre7855** [reply]
I have just disabled all extensions except UBOL, and restarted Chrome, but the problem is always here.

**Reply by stephenhawk8054** [reply]
Does it happen if you disable `AdGuard Français` list?

**Reply by Pierre7855** [reply]
I have just disabled "adguard français" list, but the problem is always here.

**Reply by stephenhawk8054** [reply]
Can you record video how the issue looks like?

**Reply by garry-ut99** [reply]
Also provide information whether the issue:
- occurs as a Guest or when Logged in, or in both cases
- still occurs in Optimal and Basic modes, and if still occurs
- then also try with "Ads, trackers, miners, and more" disabled

**Reply by Yuki2718** [reply]
Can possibly be this staff?
```
www.youtube.com##+js(nano-stb, resolve(1), *, 0.001)
www.youtube.com##+js(set, yt.config_.EXPERIMENT_FLAGS.web_enable_ab_rsp_cl, false)
www.youtube.com##+js(set, yt.config_.EXPERIMENT_FLAGS.ab_pl_man, false)
||googlevideo.com/videoplayback$xhr,3p,method=get,domain=www.youtube.com
```

**Reply by stephenhawk8054** [reply]
> Can possibly be this staff?

Yeah I suspect that too. But need video recording to be sure how it looks like.

**Reply by Pierre7855** [reply]
I have just made the video in MP4. Please see in attachment.
You can see that before the main video, there is 2 ad videos (see the time at the bottom).

https://github.com/user-attachments/assets/4783a96f-93ca-4dc3-9c48-e8384080dd3b

**Reply by stephenhawk8054** [reply]
Which uBOL version are you using?

**Reply by Pierre7855** [reply]
My UBOL version is :  uBlock Origin Lite 2024.8.21.996

**Reply by stephenhawk8054** [reply]
Hmm... I'm not sure what's wrong so far. This looks like the lack of `json-prune-fetch-response` but that version already includes the filter.

**Reply by Pierre7855** [reply]
About my problem on Youtube with UBO Lite, I have noticed that, if I uncheck the default filter "ads, trackers, miners and more", it works (for some videos, I can access directly to the main video). 
Also, as you have noticed in the video, I was logged in for Youtube.

**Comment by stephenhawk8054** [comment]
Since this is related to uBOL. I'll transfer to uBOL's repo.

**Comment by Pierre7855** [comment]
Hi @Yuki2718 , @stephenhawk8054 , @garry-ut99 ,

The problem in Youtube seems to be with the UBOL default filter ("ads, trackers, miners and more") : if I uncheck it, the result is : 
- for some youtube videos : I can access directly to the main video
- but for some other videos : I can see 2 ad video before the main video

Have you a solution that could work for all videos in Youtube, in order to access directly to the main video ?

**Reply by stephenhawk8054** [reply]
I can't find a solution if I can't reproduce the issue

**Reply by Pierre7855** [reply]
Perhaps could you send me a filter that I could enter in the software in order to test it on my side and see if that works ?

**Reply by stephenhawk8054** [reply]
I can't reproduce the issue and I don't know what causes the issue. How can I craft any filters to test?

Also uBOL cannot use custom filters.

**Reply by Pierre7855** [reply]
Ok Stephen, I understand.

In order to summarize my problem  : 
1) With "ads, tracker, miners and more" checked : 
- for all videos : I can see 2 ad videos before the main video.


2) with "ads tracker, miners and more" unchecked : 
- for some videos : it works, ie : I can access directly to the main video.
- but for some other videos : I can see 2 ad video before the main video.

**Reply by stephenhawk8054** [reply]
In case 1, do the ads still show black screen or they play as normal videos?

**Reply by Pierre7855** [reply]
in case 1, the ads are played with a black screen.
